html/pfappserver/root/src/components/new/BaseInputSwitch.vue

```diff
@@ -0,0 +1,287 @@
+<template>
+  <div class="base-input-switch-wrapper">
+    <div
+      class="base-input-switch"
+      :class="{
+        'is-focus': isFocus,
+        'is-blur': !isFocus,
+        'is-disabled': isLocked,
+        'size-sm': size === 'sm',
+        'size-md': size === 'md',
+        'size-lg': size === 'lg',
+        'switch-enabled' : rangeValue === 1,
+      }"
+      :index="inputValue"
+      @click.stop
+    >
+      <div style="pointer-events: none;">
+        <span :class="[
+          'handle',
+          rangeValue === 1 ? 'handle-on' : 'handle-off'
+        ]">
+          <icon v-if="isLocked" name="lock"/>
+          <icon v-else-if="icon" :name="icon"/>
+          <slot v-else/> <!-- Icon slot -->
+        </span>
+      </div>
+      <input type="range" ref="input"
+             :tabIndex="tabIndex"
+             :value="rangeValue"
+             :step="1"
+             :max="1"
+             :min="0"
+             :disabled="isLocked"
+             @mouseup="onClick"
+             @focus="onFocus"
+             @blur="onBlur"
+      />
+    </div>
+  </div>
+</template>
+<script>
+import {useInput, useInputProps} from '@/composables/useInput'
+import {useInputMeta, useInputMetaProps} from '@/composables/useMeta'
+import {useInputValidatorProps} from '@/composables/useInputValidator'
+import {setFormNamespace, useInputValue, useInputValueProps} from '@/composables/useInputValue'
+import {computed, inject, unref} from '@vue/composition-api';
+import _ from 'lodash'
+
+export const useSwitchProps = {
+  enabledValue: {
+    default: true
+  },
+  disabledValue: {
+    default: false
+  },
+  size: {
+    type: String,
+    default: 'md',
+    validator: value => ['sm', 'md', 'lg'].includes(value)
+  },
+  value: {
+    default: null
+  },
+  disabled: {
+    type: Boolean
+  },
+  icon: {
+    type: String
+  },
+  color: {  // override default colors via JS
+    type: String
+  },
+  hints: { // dots/pills in range for hints (eg: [1, [1-2], 2])
+    type: Array,
+    default: () => ([])
+  }
+}
+
+export const props = {
+  ...useInputProps,
+  ...useInputMetaProps,
+  ...useInputValidatorProps,
+  ...useInputValueProps,
+  ...useSwitchProps,
+}
+
+export const setup = (props, context) => {
+
+  const form = inject('form')
+
+  const metaProps = useInputMeta(props, context)
+
+  const {
+    placeholder,
+    readonly,
+    tabIndex,
+    text,
+    type,
+    isFocus,
+    isLocked,
+    onFocus,
+    onBlur
+  } = useInput(metaProps, context)
+
+  const {
+    value,
+  } = useInputValue(metaProps, context)
+
+  const onChange = (value) => {
+    setFormNamespace(props.namespace.split('.'), unref(form), value)
+  }
```

Comment:
Ideally, low-level components like this should only be tracking their state. The switch should only track whether it's ON or OFF. Instead of namespace prop, the parent component should pass on change event. The boundary with the back end (view component) can then decide what true/false means (map true to "yes", "enabled" or "1")

Reply:
I haven't made that change because it would be inconsistent. All other low-level components track and modify the store. We can make the change, but we then have to make an effort to refactor other components as well, which seems to be a colossal effort